[Java.Interop] Add JavaTypeManager class based on JavaObject

radekdoulik opened the pull request:
And use it from Java side in place of the old
`Java.Interop.TypeManager` and its `Activate` method. As it is based
on `JavaObject`, it will work nicely with the *jnimarshalmethod-gen*.

jonpryor commented on src/Mono.Android/Java.Interop/TypeManager.cs:
```diff
@@ -358,5 +358,19 @@ public static void RegisterPackages (string[] packages, Converter<string, Type>[
 			}
 		}
+
+		public class JavaTypeManager : JavaObject
```
Can this type be `internal` instead of `public`?

radekdoulik replied:
Hopefully yes. I think it should be only used from Mono.Android and the generated code, which should end up in the Mono.Android as well.

I will test whether it will indeed work in the real apk.

jonpryor replied:
It better work, or we'd implicitly be requiring that *all* `Java.Lang.Object` subclasses in all assemblies be `public`, which is *not* a pre-existing requirement. :-)

radekdoulik replied:
tested and works OK here. with new marshal methods and in fallback as well.

jonpryor commented:
Out of curiosity, why inherit `JavaObject` and not `Java.Lang.Object`? (I'm not asking that we change it; I'm just curious.)

radekdoulik commented:
Good question. Actually I don't remember :-) So I guess no particular reason. Thus I rebased it on `Java.Lang.Object` and added `RegisterAttribute` to the class, so that an unwanted *Acw* is not generated.

jonpryor commented:
On-device unit tests weren't executed on the [**macOS+xbuild PR Build** run](https://jenkins.mono-project.com/job/xamarin-android-pr-builder/2880/), so I restarted it to ensure we didn't break runtime tests.
